pkg/controller/dgraph/models/process.go

```diff
@@ -38,29 +39,33 @@ type Proc struct {
 	Name      string    `json:"name,omitempty"`
 	Interacts []*Pod    `json:"interacts,omitempty"`
 	Container Container `json:"container,omitempty"`
+	StartTime time.Time `json:"startTime,omitempty"`
+	EndTime   time.Time `json:"endTime,omitempty"`
 }
 
-func newProc(procXID, procName, containerUID string) (*api.Assigned, error) {
+func newProc(procXID, procName, containerUID, containerXID string, creationTimeStamp time.Time) (*api.Assigned, error) {
 	newProc := Proc{
 		ID:        dgraph.ID{Xid: procXID},
 		IsProc:    true,
 		Name:      procName,
-		Container: Container{ID: dgraph.ID{UID: containerUID}},
+		Container: Container{ID: dgraph.ID{UID: containerUID, Xid: containerXID}},
+		StartTime: creationTimeStamp,
 	}
 	bytes := utils.JSONMarshal(newProc)
 	return dgraph.MutateNode(bytes)
 }
 
 // StoreProcess ...
-func StoreProcess(procXID, procName string, podsXIDs []string, containerXID string) error {
+func StoreProcess(procXID, procName string, podsXIDs []string, containerXID string,
+	creationTimeStamp time.Time) error {
```

Comment:
We are passing to many arguments to the function, can we instead group a few logical ones together into a struct and pass the struct value.
Eg: `procXID` and `procName` could be grouped into proc struct.